### ISSUE TITLE
Can we connect json-server to DB? (SQLite, MySQL, etc)

### ISSUE THREAD

**Issue by asoneji** [typicode/json-server#729]
Can we connect json-server to DB? (SQLite, MySQL, etc)

We see that json-server uses [lowdb](https://github.com/typicode/lowdb) - lowdb doesn't support Cluster and may have issues with very large JSON files (~200MB).

For our testing we need to send large data files and so we wanted to use a DB instead of lowdb. Any suggestion for this issue would be a big help.

**Comment by mariokad** [comment]
^ Completely agree. It seems like many people want to do this. Any help/suggestions would be great!

**Comment by maferland** [comment]
Would be interested as well

**Comment by antonpegov** [comment]
That is funny, why you need json-server than, just use normal server )))

**Comment by asoneji** [comment]
Because json-server provides full fake REST API and we would like leverage all those capabilities.

**Comment by antonpegov** [comment]
> Because json-server provides full fake REST API and we would like leverage all those capabilities.

Really, can you imagine what a headache this task can bring to the author? All this database connectors, api upgrades, blablabla - no way. This project is just perfect.

**Comment by pantchox** [comment]
@antonpegov i agree this project is perfect as it is, but there is a potential using db connectors. 
What i mean is that whats amazing in json-server is that you can take any JSON and it will create auto express route for the CRUD operations, just from the JSON file structure - that is incredible.
So as i look it, i would not mind have a json file that describe my db structure and it will create auto routes and will connect to my db instead lowDB, that will save me TON of work creating API endpoints for each thing, and if i change the JSON file and reload the server again it will auto create my routes again with no effort.

For that i see it as a great feature @typicode any chances of this happening? :)

**Comment by drsounds** [comment]
Does json-server expose ability to hook into GET, POST, DELETE, UPDATE operations where you could add your own code or plugins that faciliates what the author intended to do?
